anoncreds/src/services/verifier.rs

```diff
@@ -675,28 +675,26 @@ fn gather_filter_info(referent: &str, identifiers: &HashMap<String, Identifier>)
         )
     })?;
 
-    let (_, schema_issuer_did, schema_name, schema_version) =
-        identifier.schema_id.parts().ok_or_else(|| {
-            err_msg!(
-                "Invalid Schema ID `{}`: wrong number of parts",
-                identifier.schema_id.0
-            )
-        })?;
+    // TODO: how can we get these as as we can not extract them from the ID anymore
+    let schema_name = String::from("");
+    let schema_version = String::from("");
+    let schema_issuer_did = String::from("");
+    let cred_def_issuer_did = Some(String::from(""));
```

Comment:
We can extract the schema_name and schema_version from the passed schemas in the `verify_presentation` method and pass them down (`verify_presentation`  -> `verify_requested_restrictions` -> `gather_filter_info`).

For the other ones, maybe we need to look at adding an `issuerId` property to anoncreds objects after all? I think there's no harm in it, and it would be backwards compatible with current anoncreds (issuerId is always the indy did in that case). Relevant issue: https://github.com/hyperledger/anoncreds-spec/issues/102

For the indy method it would be the indy did, for cheqd it would be the cheqd did. For non ledger based methods there must be some point of reference of who created the schema/credential definition. For https it could be the host (e.g. schema `https://hyperledger.org/membershipSchema` would have `issuerId` of `https://hyperledger.org`

@swcurran thoughts?

Reply:
I agree it makes sense to attach an issuer ID to the schema and cred def objects. While the ID of the object might not be known prior to publishing, the issuer ID should always be known.

Reply:
Would it be good to pick this up when the specification is updated? Seems like an annoying blocker and for now we could merge it with the TODO. I can create an issue for it to resolve it.

Reply:
I created an issue for this, will fix this once we have a concrete answer in the spec.